src/Simplify_Add.cpp

```diff
@@ -74,6 +74,7 @@ Expr Simplify::visit(const Add *op, ExprInfo *bounds) {
              rewrite(select(x, y, z) + (w + select(x, u, v)), select(x, y + u, z + v) + w) ||
              rewrite(select(x, y, z) + (select(x, u, v) - w), select(x, y + u, z + v) - w) ||
              rewrite(select(x, y, z) + (w - select(x, u, v)), select(x, y - u, z - v) + w) ||
+             rewrite(select(x, c0 - y, c1) + c2, select(x, fold(c0 + c2) - y, fold(c1 + c2)), !overflows(c0 + c2) && !overflows(c1 + c2)) ||
```

Comment:
Rules fail to match if a fold on the RHS overflows, so the predicate is unnecessary